test/core/device/intrinsics/math.jl

```diff
@@ -97,6 +104,18 @@ using SpecialFunctions
         # JuliaGPU/CUDA.jl#1085: exp uses Base.sincos performing a global CPU load
         @test testf(x->exp.(x), [1e7im])
     end
+    
+    @testset "Real - $op" for op in (exp, abs, abs2, exp10, log10)
+        @testset "$T" for T in (Float16, Float32, Float64)
+            @test testf(x->op.(x), rand(T, 1))
+        end
+    end
+    
+    @testset "Float16 - $op" for op in (log,exp,exp2,exp10,log2,log10)
+        @testset "$T" for T in (Float16, )
+            @test testf(x->op.(x), rand(T, 1))
+        end
+    end
```

Comment:
We could test all values here:

```
julia> all_float_16 = collect(reinterpret(Float16, pattern) for pattern in  UInt16(0):UInt16(1):typemax(UInt16))
65536-element Vector{Float16}:
```
(there might be a better way that avoids some of the duplicated patterns, but it is only 65k in the end)

Otherwise for some of the degenerate cases we might randomly fail if we disagree with Julia.

Reply:
```
all_float_16 = collect(reinterpret(Float16, pattern) for pattern in  UInt16(0):UInt16(1):typemax(UInt16))
all_float_16 = filter(!isnan, all_float_16)

julia> findall(==(0), exp.(all_float_16) .== Array(exp.(CuArray(all_float_16))))
2-element Vector{Int64}:
 8058
 9680

julia> all_float_16[8058]
Float16(0.007298)

julia> all_float_16[9680]
Float16(0.02269)

julia> reinterpret(UInt16, all_float_16[8058])
0x1f79

julia> reinterpret(UInt16, all_float_16[9680])
0x25cf
```